Eval fixes for cross-compilation

Mindavi opened the pull request:
###### Description of changes

Fix evaluation of `nix-env -f . -qaP --attr pkgsCross.aarch64-multiplatform --out-path --show-trace` for a bunch of packages. It still doesn't fully evaluate, unfortunately.

Improves progress for https://github.com/NixOS/nixpkgs/issues/116207.

###### Things done

- Tried evaluating nixpkgs with the above function


Opening as draft for now since I'm still working at it, but I like sharing it already :)

Mindavi committed:
appvm: makeWrapper to nativeBuildInputs

Mindavi committed:
acl2: makeWrapper to nativeBuildInputs

Mindavi committed:
apache-jena: makeWrapper to nativeBuildInputs

Mindavi committed:
azure-functions-core-tools: some deps to nativeBuildInputs

Mindavi committed:
arion: use build makeWrapper instead of host one

Mindavi committed:
chickenPackages_4.chicken, chickenPackages_5.chicken: move makeWrapper to nativeBuildInputs

jshcmpbll reviewed (commented):
```zsh
mbazg40bpcmixjin0x2jxynnd4kram9q-azure-functions-core-tools-3.0.3785/bin ./func
Process terminated. Couldn't find a valid ICU package installed on the system. Set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support.
   at System.Environment.FailFast(System.String)
   at System.Globalization.GlobalizationMode.GetGlobalizationInvariantMode()
   at System.Globalization.GlobalizationMode..cctor()
   at System.Globalization.CultureData.CreateCultureWithInvariantData()
   at System.Globalization.CultureData.get_Invariant()
   at System.Globalization.CultureInfo..cctor()
   at System.StringComparer..cctor()
   at System.StringComparer.get_OrdinalIgnoreCase()
   at Azure.Functions.Cli.Program.SetCoreToolsEnvironmentVariables(System.String[])
   at Azure.Functions.Cli.Program.Main(System.String[])
[1]    95172 abort (core dumped)  ./func
```
Looks like ive got some other things to fix in this derivation 😞

SuperSandro2000 reviewed (commented):
LGTM except I can't comment on the buildPackages part.

Mindavi committed:
cling: use makeWrapper for build platform

Mindavi committed:
asls: move erlangR22 to nativeBuildInputs

Mindavi committed:
varnish: move makeWrapper to nativeBuildInputs

Mindavi committed:
conda: use makeWrapper for build platform

Mindavi committed:
couchdb: fix eval for cross-compilation

Use erlangR22 for the build platform.
Requires buildPackages in top-level due to no splicing at that level.

Mindavi commented:
Let's leave it at this for now, it appears there's quite some things to go through, so let's not make this enormous. (Besides, it takes a lot of time and RAM to do an eval already).

Mindavi commented:
Will need to drop a commit (the one for build-dotnet-module) due to this being merged: https://github.com/NixOS/nixpkgs/pull/173889

Artturin commented on pkgs/tools/package-management/conda/default.nix:
```diff
@@ -42,7 +42,7 @@ let
       ];
     in
-      runCommand "conda-install" { buildInputs = [ makeWrapper zlib]; }
+      runCommand "conda-install" { buildInputs = [ buildPackages.makeWrapper zlib]; }
```
`nativeBuildInputs` works in `runCommand`

Artturin commented:
steam eval fix in https://github.com/NixOS/nixpkgs/pull/186512

Mindavi commented:
Thanks for rebasing :)

Artturin commented:
more https://github.com/NixOS/nixpkgs/pull/186564
